torch/_inductor/ir.py

```diff
@@ -4676,6 +4714,64 @@ def require_stride1(cls, x):
                     return x
         return cls.copy_input(x)
 
+    @classmethod
+    def require_exact_strides(cls, x, exact_strides, allow_padding=False):
+        assert exact_strides is not None
+        order = get_stride_order(exact_strides)
+        if x.get_numel() == 0:  # Layout doesn't matter
+            return x
+        if is_storage_and_layout(x):
+            while isinstance(x.get_layout(), NonOwningLayout):
+                x = x.get_layout().view
+            if isinstance(x.get_layout(), FlexibleLayout):
+                as_storage_and_layout(
+                    x,
+                    freeze=True,
+                    want_contiguous=False,
+                    stride_order=None,
+                    allow_padding=allow_padding,
+                    exact_strides=exact_strides,
+                )
+                return x
+            # @TODO: do we need to check `x.get_layout().is_stride_ordered(order)`?
+            elif isinstance(x.get_layout(), FixedLayout) and list(
+                x.get_layout().stride
+            ) == list(exact_strides):
+                return x
+            elif isinstance(x.get_layout(), MutationLayoutSHOULDREMOVE):
+                if isinstance(x.get_layout().real_layout(), FlexibleLayout):
+                    raise AssertionError(
+                        "the MutationLayoutSHOULDREMOVE's real layout shouldn't be FlexibleLayout"
+                    )
+                elif isinstance(x.get_layout().real_layout(), FixedLayout):
+                    return x
+        # TODO - Storage to InputBuffer
+        if isinstance(x, InputBuffer) and x.get_layout().is_stride_ordered(order):
+            return x
+        if (
+            isinstance(x, TensorBox)
+            and isinstance(x.data, BaseView)
+            and not isinstance(x.data, ReinterpretView)
+            and is_storage_and_layout(x.unwrap_view())
+            and not isinstance(x.unwrap_view().data, ExternKernelAlloc)
+        ):
```

Comment:
This is a copy pasta of require_stride_order. We should not duplicate code. Can we refactor ?

Reply:
yeah. I think so. let me rename the original api.